pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,673 @@
+			err := client.Secrets.Drop(ctx, sdk.NewDropSecretRequest(id).WithIfExists(true))
+			require.NoError(t, err)
+		}
+	}
+
+	createSecretWithOAuthClientCredentialsFlow := func(t *testing.T, integrationId sdk.AccountObjectIdentifier, scopes []sdk.SecurityIntegrationScope, with func(*sdk.CreateWithOAuthClientCredentialsFlowSecretRequest)) (*sdk.Secret, sdk.SchemaObjectIdentifier) {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithOAuthClientCredentialsFlowSecretRequest(id, integrationId, scopes)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithOAuthClientCredentialsFlow(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	createSecretWithOAuthAuthorizationCodeFlow := func(t *testing.T, integrationId sdk.AccountObjectIdentifier, refreshToken, refreshTokenExpiryTime string, with func(*sdk.CreateWithOAuthAuthorizationCodeFlowSecretRequest)) (*sdk.Secret, sdk.SchemaObjectIdentifier) {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithOAuthAuthorizationCodeFlowSecretRequest(id, refreshToken, refreshTokenExpiryTime, integrationId)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithOAuthAuthorizationCodeFlow(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	createSecretWithBasicAuthentication := func(t *testing.T, username, password string, with func(*sdk.CreateWithBasicAuthenticationSecretRequest)) (*sdk.Secret, sdk.SchemaObjectIdentifier) {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithBasicAuthenticationSecretRequest(id, username, password)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithBasicAuthentication(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	createSecretWithGenericString := func(t *testing.T, secretString string, with func(options *sdk.CreateWithGenericStringSecretRequest)) (*sdk.Secret, sdk.SchemaObjectIdentifier) {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithGenericStringSecretRequest(id, secretString)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithGenericString(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	assertSecret := func(t *testing.T, s *sdk.Secret, expectedId sdk.SchemaObjectIdentifier, expectedSecretType, expectedComment string) {
```

Comment:
We are steering away from those kinds of assert functions because now we have generated ones. Please try to generate them and use them here.